miniapps/dpg/util/pweakform.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_PDPGWEAKFORM
+#define MFEM_PDPGWEAKFORM
+
+#include "mfem.hpp"
+#include "weakform.hpp"
+
+#ifdef MFEM_USE_MPI
+
+namespace mfem
+{
+
+/** @brief Class representing the whole weak formulation. (Convenient for DPG or
+    Normal Equations) */
+class ParDPGWeakForm : public DPGWeakForm
+{
+
+protected:
+   // Trial FE spaces
+   Array<ParFiniteElementSpace * > trial_pfes;
+
+   // ess_tdof list for each space
+   Array<Array<int> *> ess_tdofs;
+
+   // split ess_tdof_list give in global tdof (for all spaces)
```

Comment:
`give` -> `given`. The comment `this can be changed...` is not clear to me.

Reply:
fixed